src/schema/v2/fields/money.ts

```diff
@@ -137,6 +137,18 @@ export const Money = new GraphQLObjectType<any, ResolverContext>({
         return cents / factor
       },
     },
+    toUSD: {
+      type: new GraphQLNonNull(GraphQLFloat),
+      description: "An amount of money converted to USD major units (dollars).",
+      resolve: async ({ cents, currency }, {}, { exchangeRatesLoader }) => {
+        const factor = currencyCodes[currency.toLowerCase()].subunit_to_unit
+        const major = cents / factor
+
+        const exchangeRates = await exchangeRatesLoader()
```

Comment:
On the one hand, I'm thinking that while we generally don't want to swallow errors it seems as though an error fetching an exchange rate, or an exchange rate not being found for a particular currency, should return `null` vs. an error?

However, with recent work in Reaction/Force to not crash the app based on _any_ error resolving a leaf node of a GraphQL query, vs. just a very short list of crash-able errors, that may be less important (ie- our UI's _should_ be resilient to these kinds of errors).

Reply:
Hm I'm not up on those latest error-handling changes. Is there a specific change you'd prefer to see here?

Reply:
Yea, basically invoking `exchangeRatesLoader` might throw an error, or the corresponding exchange rate might be not found, so you'd have a divisor of `null` when calculating the USD amount.

Basically wondering if in either/both of those cases, you may want to explicitly be defensive and return `null`, vs possibly throwing an error.

But, `null` feels like an odd return type, plus this is a non nullable field.

So I think this might be good as-is!

Reply:
Gotcha, yeah I guess raising an error on a problem with loading exchange rates sounds appropriate then. Is there anything I need to know about the new Force error handling with regard to this? Is this the "principal field" work?

Reply:
Yea, basically we used to throw an error (which server-side would be a 500 error page to the user), on _any_ GraphQL `error` being present....even if the only error for a large artwork page query was one teensy leaf, an exchange rate error'ing for instance.

So we got used to being defensive about those scenarios. And not that it's necessarily bad to be defensive! But, that defensiveness might be better in the UI component. And, with the `principalField` directive and some updates to Reaction middleware, we now only throw away the query if an error occurs in a `principalField` resolver, otherwise we let the components render.

So just for sanity and peace of mind it might have been better to be defensive everywhere- Metaphysics, UI's, etc. However, now we can more properly report leaf errors and let the UI decide, and no longer need to be so overly defensive in the MP layer.

* sorry for that long-winded explanation! I owe a blog post!